main/sceneManager.cpp

```diff
@@ -1,26 +1,66 @@
 #include "main/sceneManager.h"
 
-using Tri = objects::TriangleObj;
+SceneManager::SceneManager()
+{
+    createCollectors();
+}
+
+std::vector<objects::EnergyCollector *> SceneManager::getEnergyCollectors()
+{
+    std::vector<objects::EnergyCollector *> temp;
+    temp.reserve(_energyCollectors.size());
+    for (const auto &t : _energyCollectors)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
+
+std::vector<objects::TriangleObj *> SceneManager::getModelTriangles()
+{
+    std::vector<objects::TriangleObj *> temp;
+    temp.reserve(_modelTriangles.size());
+    for (const auto &t : _modelTriangles)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
 
-void IamSimulator()
+// PRIVATE METHODS
+
+void SceneManager::createCollectors()
 {
-    SceneManager manager;
-    CHECK(manager.loadModel("123231")) << "Cannot load model";
+    _energyCollectors.reserve(constants::kCollectors);
 
-    std::vector<Tri> triangles = manager.getModelTriangles();
-    generators::PointSource source;
+    const double collectorRadius = static_cast<double>(constants::kSimulationRadius) / 2;
+    double alphaIncrement;
 
-    for (int i = 0; i < 1000000; i++)
+    if (constants::kCollectors % 2 == 0)
+    {
+        alphaIncrement = 2 * constants::kPi * 180 / (static_cast<double>(constants::kCollectors) / 2 * 360);
```

Comment:
Issues:
1. It looks to me like this mixes radians (pi) and degrees (360, 180). Rewrite the formula so that you use either one of those.
2. Rewrite the IF statement so that the formula is written only once. There is no need to use an IF statement of any kind, and you can write this in one line (how to do it is an exercise to the reader).

Reply:
noted and will try... This method doesn't work anyway, so I need to rewrite it again later :P